flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/rules/logical/PushFilterIntoTableSourceScanRule.scala

```diff
@@ -68,7 +68,7 @@ class PushFilterIntoTableSourceScanRule extends RelOptRule(
       RexProgramExtractor.extractConjunctiveConditions(
         program,
         call.builder().getRexBuilder,
-        new FunctionCatalog("default_catalog", "default_database"))
+        new FunctionCatalog(null))
```

Comment:
are we able to pass in CatalogManager here?

Reply:
found that this is taken care of as part of https://github.com/apache/flink/pull/8782

Reply:
I think there's a problem with the previous code here. There must be no function in the freshly produced `FunctionCatalog`.
But if pass null to FunctionCatalog, there will be a NPE.
How about pass a `new CatalogManager(defaultCatalogName, new GenericInMemoryCatalog)`?